client/typescript/fine-uploader.d.ts

```diff
@@ -1,2975 +1,2306 @@
+         *
+         * ###Note:
+         * Enabling this will disable multiple file selection
+         *
+         * @default `false`
+         */
+        ios?: boolean;
     }
 
     /**
-     * onComplete function type
+     * ConcurrentOptions
      */
-    interface OnComplete {
+    export interface ConcurrentOptions {
         /**
-         * @param number id : The current file's id
-         * @param string name : The current file's name
-         * @param Object responseJSON : The raw response from the server
-         * @param XMLHttpRequest xhr : The object used to make the request
+         * Allow multiple chunks to be uploaded simultaneously per file
+         *
+         * @default `false`
          */
-        (id: number, name: string, responseJSON: any, xhr: XMLHttpRequest): void;
+        enabled?: boolean;
     }
 
     /**
-     * onAllComplete function type
+     * ChunkingOptions
      */
-    interface OnAllComplete {
+    export interface ChunkingOptions {
         /**
-         * @param number[] succeeded : IDs of all files in the group that have uploaded successfully (status = `qq.status.UPLOAD_SUCCESSFUL`)
-         * @param number[] failed : IDs of all files in the group that have failed (status = `qq.status.UPLOAD_FAILED`)
+         * concurrent Chunking options
          */
-        (succeeded: number[], failed: number[]): void;
-    }
-
-    /**
-     * onDelete function type
-     */
-    interface OnDelete {
+        concurrent?: ConcurrentOptions;
         /**
-         * @param number id : The current file's id
+         * Enable or disable splitting the file separate chunks. Each chunks is sent in a separate requested
+         *
+         * @default `false`
          */
-        (id: number): void;
-    }
-
-    /**
-     * onDeleteComplete function type
-     */
-    interface OnDeleteComplete {
+        enabled?: boolean;
         /**
-         * @param number id : The current file's id
-         * @param XMLHttpRequest xhr : The object used to make the request
-         * @param boolean isError : `true` if there has been an error, `false` otherwise
+         * Ensure every file is uploaded in chunks, even if the file can only be split up into 1 chunk.
+         *
+         * Does not apply if chunking is not possible in the current browser
+         *
+         * @default `false`
          */
-        (id: number, xhr: XMLHttpRequest, isError: boolean): void;
-    }
-
-    /**
-     * onError function type
-     */
-    interface OnError {
+        mandatory?: boolean;
         /**
-         * @param number id : The current file's id
-         * @param string name : The current file's name
-         * @param string errorReason : The reason for the current error
-         * @param XMLHttpRequest xhr : The object used to make the request
+         * The maximum size of each chunk, in bytes
+         *
+         * @default `200000
```

Comment:
@jclangst I see lot of function declarations have been changed to use constructors.
Is there a specific problem with previous declarations?